aws-codeguruprofiler-profilinggroup/aws-codeguruprofiler-profilinggroup.json

```diff
@@ -80,13 +92,18 @@
     },
     "update": {
       "permissions": [
+        "sns:Publish",
```

Comment:
There is no SNS publish action being called anywhere...

Reply:
You're right, but after the CR revision and upon creating a notification channel, sns:Publish is called as part of `AddNotificationChannels` call.

Reply:
Lambros is correct here: it doesn't really matter what happens inside our service (notice that there's also no calls to DynamoDB or any other services we may make use in the backend), the permissions here are for the code that you see in this package only, and that code does not need SNS permissions because it does not call SNS.

(Thanks Lambros for the great insight :1st_place_medal: :100: 🎯)

Reply:
When you don't explicitly add these, you get ""Invalid request provided: AWS::CodeGuruProfiler::ProfilingGroupRequester doesn't have permissions to publish on topicArn: <ARN> (Service: CodeGuruProfiler, Status Code: 400, Request ID: <REQ_ID>, Extended Request ID: null)""

And after tracing it, it traces to the backend code that throws that error, which I will not post here :)

Reply:
So the difference here is that for those DDB calls, those are happening in our backend, so using our service account creds. On the other hand, for the SNS::Publish, we are using the customers creds via CFN's assumedRole, which is why we have to explicitly enable the SNS:Publish on the customer's behalf

Reply:
I will leave approval and comments to Ivo but just to add the customer side.

I think there is some misconfiguration here regarding this, or maybe the docs need updating as well. Imagine that I do the `AddNotificationChannel` action through the Console, does my user/role need to have `sns:Publish` permissions as well or just the `codeguru-profiler:AddNotificationChannels` permission?

There are arguments for both and it makes sense that someone needs to have `sns:Publish` permissions to the topic they are trying to add notifications otherwise people could subscribe any topic to some noisy profiling group and cause problems. However, on the AWS docs for the [`codeguru-profiler:AddNotificationChannels`](https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_AddNotificationChannels.html) there is nothing related to this apart from the [`eventPublisher`](https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_Channel.html#profiler-Type-Channel-eventPublishers) which is a CodeGuru thing and does not mention the SNS permissions either.

It needs to be clarified otherwise people **will hit issues with this and trying to figure out what's wrong**. Remember that CloudFormation is just another client of the API so any weird issue found here will be found by manually using the CLI/Console as well.

So, my question boils down to: Is the `sns:Publish` permission needed here to avoid having the user to have it, or is it needed here AND by the user permissions as well? (I remember CloudFormation taking the intersection of user permissions and service role permissions but maybe I remember wrong on this)

ps. The [permissions docs](https://docs.aws.amazon.com/codeguru/latest/profiler-ug/auth-and-access-control-permissions-reference.html) seem to be outdated?

Reply:
Agreed, I will as an action item make sure we communicate this in our docs with Nichole, because this is unclear and this confused me and other devs on the team a lot before we realized that it was necessary.